### PR TITLE
Allow customers to set custom resource attributes as container Tags

### PR DESCRIPTION
**Pull request by dineshg13**
Currently users of DDconnector can only set fixed set of resource attributes as container Tags , list is defined [here](https://github.com/DataDog/opentelemetry-mapping-go/blob/ba4072f84888749a00f5748cca9f2a7fc2e42fc9/pkg/otlp/attributes/attributes.go#L41) . This PR enables the customers to choose any resource attribute to be set as container Tag on the APM Trace metrics payload.

### CODE REVIEW

**Review by songy23 — APPROVED** [review]
LGTM except minor nits

File: connector/datadogconnector/connector_test.go [hunk 1]
```diff
@@ -166,8 +168,8 @@ func TestContainerTags(t *testing.T) {
 	require.NoError(t, err)
 
 	tags := sp.Stats[0].Tags
-	assert.Equal(t, 2, len(tags))
-	assert.ElementsMatch(t, []string{"region:my-region", "zone:my-zone"}, tags)
+	assert.Equal(t, 3, len(tags))
+	assert.ElementsMatch(t, []string{"region:my-region", "zone:my-zone", "az:my-az"}, tags)
```
Comment (jdgumz) [inline connector/datadogconnector/connector_test.go:172]: 👍

File: .chloggen/dinesh.gurumurthy_expand-resource-attributes.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+component: datadog/connetor
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Enable connetor to use any attribute from the resource as Container Tag
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [32224]
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |
+  This change allows the connector to use any attribute from the resource as a container tag. This is useful when you want to use a custom attribute from the resource as a container tag. For example, you can use the `namespace` attribute from the resource as a container tag. This change is backward compatible and does not require any changes to the existing configuration.
```
Comment (mx-psi) [inline .chloggen/dinesh.gurumurthy_expand-resource-attributes.yaml:15]: The last part is unnecessary: if it were a breaking change then the `change_type` would be `breaking`, not `enhancement`.
```suggestion
  This change allows the connector to use any attribute from the resource as a container tag. This is useful when you want to use a custom attribute from the resource as a container tag. For example, you can use the `namespace` attribute from the resource as a container tag.
```

File: .chloggen/dinesh.gurumurthy_expand-resource-attributes.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: datadog/connetor
```
Comment (mx-psi) [inline .chloggen/dinesh.gurumurthy_expand-resource-attributes.yaml:6]:
```suggestion
component: datadog/connector
```

File: .chloggen/dinesh.gurumurthy_expand-resource-attributes.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Enable connetor to use any attribute from the resource as Container Tag
```
Comment (mx-psi) [inline .chloggen/dinesh.gurumurthy_expand-resource-attributes.yaml:8]:
```suggestion
note: Enable connector to use any attribute from the resource as Container Tag
```

File: connector/datadogconnector/connector.go [hunk 5]
```diff
@@ -81,7 +81,17 @@ func newTraceToMetricConnector(set component.TelemetrySettings, cfg component.Co
...
-	ddtags := attributes.ContainerTagFromAttributes(ctags)
+	ddTags := attributes.ContainerTagFromAttributes(ctags)
+
+	for key := range ctags {
+		// TODO: we need to expose the container mappings from the attributes package
+		tags := attributes.ContainerTagFromAttributes(map[string]string{key: ""})
+		// It means there is no mapping for the resource attribute
+		if len(tags) == 0 {
+			ddTags[key] = ""
+		}
```
Comment (mx-psi) [inline connector/datadogconnector/connector.go:92]: We have full control over the `attributes` package, we don't need to resort to hacks, we can just file a PR over there. I can merge this PR tomorrow morning if that PR has not landed, but I will wait until the PR is open

**Review by mx-psi — CHANGES_REQUESTED** [review]

**Review by songy23 — APPROVED** [review]
LGTM

**Review by mx-psi — APPROVED** [review]

**Pull request merged**